tpp-app/src/services/cycle/CycleService.js

```diff
@@ -0,0 +1,304 @@
+      tomorrow = current;
+      current = yesterday;
+
+      twoDaysLaterSymptoms = tomorrowSymptoms;
+      tomorrowSymptoms = dateSymptoms;
+      dateSymptoms = await getSymptomsFromCalendar(calendar, current.getDate(), current.getMonth() + 1, current.getFullYear());
+      noFlowToday = (dateSymptoms.flow === FLOW_LEVEL.NONE || dateSymptoms.flow === null) ;
+      noFlowTomorrow = (tomorrowSymptoms.flow === FLOW_LEVEL.NONE || tomorrowSymptoms.flow === null) ;
+      flowTwoDaysLater = (twoDaysLaterSymptoms.flow !== null && twoDaysLaterSymptoms.flow !== FLOW_LEVEL.NONE)
+    }
+
+
+
+    //return twoDaysLater since pattern searching for is _ _ X where X is the period
+    return twoDaysLater;
+}
+
+const CycleService = {
+  /**
+   *  Store how far the user is into their period as a percentage
+   *  @param {number} percent Float in range [0,1] of how far along period is
+   *  @return {Promise} Resolves when the set operation is completed
+   */
+  POSTCycleDonutPercent: async function(percent){
+    try {
+      var today = new Date();
+      var date = getDateString(today);
+
+      let datePercent = {
+      }
+      datePercent[date] = percent;
+      return await AsyncStorage.setItem(Keys.CYCLE_DONUT_PERCENT, JSON.stringify(datePercent));
+    } catch (e) {
+      reject("Something went wrong");
+    }
+  },
+
+  /**
+   * Get the user's average period length
+   * @return {Promise} Resolves into either an integer for number of days or NULL if info not present
+   */
+  GETAveragePeriodLength: async function(){
+    try {
+      const res = await AsyncStorage.getItem(Keys.AVERAGE_PERIOD_LENGTH);
+      return res;
+    } catch (e) {
+      reject("Something went wrong");
+      console.log(e);
+      return null;
+    }
+
+  },
+  /**
+   * Get the user's average cycle length
+   * @return {Promise} Resolves into either an integer for number of days or NULL if info is not present
+   */
+  GETAverageCycleLength: async function(){
+    try {
+      const res = await AsyncStorage.getItem(Keys.AVERAGE_CYCLE_LENGTH);
+      return res;
+    } catch (e) {
+      console.log(e);
+      return null;
+    }
+
+  },
+
+  /**
+   * Get the number of days the user has been on their period
+   *
+   * @param {Object} calendar The object containing the symptoms for this year, last year, and next year. Optional.
+   * @return {Promise} Resolves into 0 if user not on period, and an integer of the days they have been on their period otherwise
+   */
+  GETPeriodDay: async function (calendar = null){
+
+    let periodDays = 0;
+    var date = new Date()
+    if (!calendar){
+      calendar = await getCalendarByYear(date.getFullYear());
+    }
+    let dateSymptoms = await getSymptomsFromCalendar(calendar, date.getDate(), date.getMonth()+1, date.getFullYear());
+    if (dateSymptoms.flow === null || dateSymptoms.flow === FLOW_LEVEL.NONE){
+      return 0;
+    }
+    else {
+      let startDate = await this.GETMostRecentPeriodStartDay();
```

Comment:
just curious if there was a reason to call `GETMostRecentPeriodStartDay` here rather than calling `getLastPeriodStart` directly with today's date via `new Date()`?

Reply:
I used `GETMostRecentPeriodStartDay` b/c it's in the defined API so I already had it. It's functionally equivalent to calling `getLastPeriodStart` directly with today's date via new Date() like u said, but shorter. Do you think we need both of the functions, since there's a pretty significant overlap in functionality?

Reply:
oo i see hmmm personally it feels a bit redundant to me to have both but i also get it if we just keep it as is